examples/react/farm.config.ts

```diff
@@ -3,21 +3,24 @@ import type { UserConfig } from '@farmfe/core';
 export default <UserConfig>{
   compilation: {
     input: {
-      index: './index.html'
+      index: './index.html',
     },
     resolve: {
-      symlinks: true
+      symlinks: true,
     },
     define: {
-      BTN: 'Click me'
+      BTN: 'Click me',
+      HMR_PORT: 8008,
```

Comment:
The hmr port should using `server.hmr.port` instead of configured by user.

Reply:
ok

Reply:
> The hmr port should using `server.hmr.port` instead of configured by user.
我需要写一个这样的类 吗  如下：
```
const HMR_CONFIG_KEY = '__hmrConfig';
class ConfigInjectionPlugin {
  private config: any;

  constructor(config: any) {
    this.config = config;
  }

  apply(moduleSystem: any) {
    moduleSystem.setExports('@farmfe/runtime-plugin-hmr', (module: any) => {
      console.log(module, '---module');
      module.exports = {
        ...module.exports,
        [HMR_CONFIG_KEY]: this.config,
      };
    });
  }
}

export default ConfigInjectionPlugin;
```

然后在 调用  require.resolve之前加载它 是吗
![image](https://user-images.githubusercontent.com/51783403/233974983-6643c4fa-c7c1-4739-9251-305d4ce00ef0.png)

Reply:
> 
这个功能会涉及到rust注入的改动吗，可以像下面写一个这样的类吗：
![fca854efe7099b2599c0137e365e08a](https://user-images.githubusercontent.com/51783403/235295750-6d22127e-3c39-4d47-b508-503339368a7f.png)
然后我不知道farm的 moduleSysdem 如何被引入到normalizeUserCompilationConfig定义的这个文件

Reply:
> > 
> 
> 这个功能会涉及到rust注入的改动吗，可以像下面写一个这样的类吗： ![fca854efe7099b2599c0137e365e08a](https://user-images.githubusercontent.com/51783403/235295750-6d22127e-3c39-4d47-b508-503339368a7f.png) 然后我不知道farm的 moduleSysdem 如何被引入到normalizeUserCompilationConfig定义的这个文件

Reply:
> The hmr port should using `server.hmr.port` instead of configured by user.